bin/pc2installservice

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+# Copyright (C) 1989-2023 PC2 Development Team: John Clevenger, Douglas Lane, Samir Ashoo, and Troy Boudreau.
+#
+# Purpose:
+#   This script installs a Linux service that runs the script to create
+#   the cgroup necessary for PC2 sandboxes.  The service is a one-shot service
+#   that simply runs the /usr/pc2/bin/pc2installsandbox shell script
+#
+# Author: John Buck
+
+#DEBUG=
+DEBUG="echo Would"
+
+SCRIPTNAME="$0"
+
+if test $# -ne 2
+then
+	echo Usage: $SCRIPTNAME pc2user pc2group
+	exit 4
+fi
+JUDGE_USER="$1"
+JUDGE_GROUP="$2"
+if ! getent passwd $JUDGE_USER >/dev/null 2>&1
+then
+	echo $SCRIPTNAME: No such user $JUDGE_USER.  The pc2user must be a valid login id.
+	exit 5
+fi
+if ! getent group $JUDGE_GROUP >/dev/null 2>&1
+then
+	echo $SCRIPTNAME: No such group $JUDGE_USER.  The pc2group must be a valid group in /etc/group.
+	exit 6
+fi
+
+INSTALL_SCRIPT_PATH=/usr/pc2/bin/pc2installsandbox
+SERVICE_NAME=pc2_cgroup.service
+SERVICE_FILE=/usr/pc2/support/system_cgroup/${SERVICE_NAME}
+
+ServiceError()
+{
+	echo $SCRIPTNAME: $*
+	echo $SCRIPTNAME: Service not installed.
+	echo $SCRIPTNAME: For help, please refer to the '"Running in a Sandbox"' Appendix
+	echo $SCRIPTNAME: in the \"PC2 Contest Administrator\'s Installation and Configuration Guide\"
+}
+
+euid=`id -u`
+if test "$euid" -ne 0
+then
+	echo $SCRIPTNAME: This script must be run as super user
+	exit 1
+fi
+if test ! -s ${SERVICE_FILE}
+then
+	ServiceError Can not find service file: ${SERVICE_FILE}
```

Comment:
This is probably a personal preference thing, but I think "Can not" in the above line should be "Cannot".   The word "cannot" indicates something which is impossible:  "I cannot file the service file".  The use of two separate words "can not" tends to indicate an option:  "I can go shopping, or I can not go shopping; it's up to you."   Again, probably just a personal thing, and I can certainly find online sources that disagree and instead define "can not" and "cannot" as meaning the same thing.  Certainly I cannot withhold approval for this reason, but I can not approve it if you wish :)